js/explosions.js

```diff
@@ -0,0 +1,47 @@
+var Explosion = function (x, y) {
+  // x and y is the location of the colliding ammo
+  this.x = x ;
+  this.y = y ;
+  this.width = 80;
+  this.height = 80;
+  this.halfWidth = (this.width / 2);
+  this.halfHeight = (this.height / 2);
+  this.srcSpriteSize = 64;
+  this.spriteIndexArr = [0,1,2,3,4];
+  this.i = 0;
+  this.numExplosionFrames = 5;
+  this.image = new Image(),
+  this.image.src = "resources/images/ExplosionStrip.png";
+  this.remove = false;
+}
+
+Explosion.prototype.xPosition = function (index) {
+  return 0;
+}
+
+Explosion.prototype.yPosition = function (index) {
+  return index;
+}
+
+Explosion.prototype.render = function(context) {
+  if (this.i < this.numExplosionFrames) {
```

Comment:
something bothers me a little bit with a property called `i`.. would you consider renaming it to what it actually represents?